presto-hive/src/test/java/com/facebook/presto/hive/parquet/AbstractTestParquetReader.java

```diff
@@ -181,6 +181,101 @@ public void testNestedArrays()
+        for (List<Integer> value : values) {
+            List<SqlDecimal> expectedDecimal = new ArrayList<>();
+            for (Integer valueInt : value) {
+                expectedDecimal.add(SqlDecimal.of(valueInt, precision, scale));
+            }
+            expectedValues.add(expectedDecimal);
+        }
+
+        MessageType hiveSchema = parseMessageType(format("message hive_list_decimal {" +
+                "  optional group my_list (LIST){" +
+                "    repeated group list {" +
+                "        optional INT32 element (DECIMAL(%d, %d));" +
+                "    }" +
+                "  }" +
+                "} ", precision, scale));
+
+        tester.testRoundTrip(objectInspector, values, expectedValues, "my_list", type, Optional.of(hiveSchema));
+    }
+
+    @Test
+    public void testNestedArraysDecimalBackedByINT64()
+            throws Exception
+    {
+        int precision = 10;
+        int scale = 2;
+        ObjectInspector objectInspector = getStandardListObjectInspector(javaLongObjectInspector);
+        Type type = new ArrayType(createDecimalType(precision, scale));
+        Iterable<List<Long>> values = createTestArrays(longsBetween(1, 1_000));
+
+        List<List<SqlDecimal>> expectedValues = new ArrayList<>();
+        for (List<Long> value : values) {
+            List<SqlDecimal> expectedDecimal = new ArrayList<>();
+            for (Long valueLong : value) {
+                expectedDecimal.add(SqlDecimal.of(valueLong, precision, scale));
+            }
+            expectedValues.add(expectedDecimal);
+        }
+
+        MessageType hiveSchema = parseMessageType(format("message hive_list_decimal {" +
+                "  optional group my_list (LIST){" +
+                "    repeated group list {" +
+                "        optional INT64 element (DECIMAL(%d, %d));" +
+                "    }" +
+                "  }" +
+                "} ", precision, scale));
+        tester.testRoundTrip(objectInspector, values, expectedValues, "my_list", type, Optional.of(hiveSchema));
+    }
+
+    @Test
+    public void testNestedArraysShortDecimalBackedByBinary()
+            throws Exception
+    {
+        int precision = 1;
+        int scale = 0;
+        ObjectInspector objectInspector = getStandardListObjectInspector(new JavaHiveDecimalObjectInspector(new DecimalTypeInfo(precision, scale)));
+        Type type = new ArrayType(createDecimalType(precision, scale));
+        ContiguousSet<BigInteger> bigIntegerValues = bigIntegersBetween(BigDecimal.valueOf(Math.pow(10, precision - 1)).toBigInteger(),
+                BigDecimal.valueOf(Math.pow(10, precision)).toBigInteger());
+        ImmutableList.Builder<HiveDecimal> writeValues = new ImmutableList.Builder<>();
+        for (BigInteger value : limit(bigIntegerValues, 1_000)) {
+            writeValues.add(HiveDecimal.create(value, scale));
+        }
+        Iterable<List<HiveDecimal>> values = createTestArrays(writeValues.build());
```

Comment:
I would consider factoring out the input and expected output into separate methods for readability.